source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -85,21 +86,27 @@ MStatus MaterialXNode::initialize()
 
 void MaterialXNode::createOutputAttr(MDGModifier& mdgModifier)
 {
-	if (materialXData)
+	if (materialXData && materialXData->getFragmentWrapper())
 	{
-		MFnNumericAttribute nAttr;
-
-		MString outputName = materialXData->glslFragmentWrapper->getFragmentName().c_str();
-		_outAttr = nAttr.createColor(outputName, outputName);
-		CHECK_MSTATUS(nAttr.setStorable(false));
-		CHECK_MSTATUS(nAttr.setInternal(false));
-		CHECK_MSTATUS(nAttr.setReadable(true));
-		CHECK_MSTATUS(nAttr.setWritable(false));
-		CHECK_MSTATUS(nAttr.setCached(true));
-		CHECK_MSTATUS(nAttr.setHidden(false));
-
-		mdgModifier.addAttribute(thisMObject(), _outAttr);
-//		CHECK_MSTATUS(addAttribute(_outAttr));
+        const MaterialX::StringMap& outputMap = materialXData->getFragmentWrapper()->getPathOutputMap();
```

Comment:
Fix the call here to use the correct one. We want the output name and not the fragment name. For now we just create 1 output, unless we can support multiple?